util/src/test/scala/HighPrecisionMoneySpec.scala

```diff
@@ -125,5 +126,23 @@ class HighPrecisionMoneySpec extends FunSpec with MustMatchers {
         )
       )
     }
+
+    it("should validate fractionDigits (min)") {
+      val Invalid(errors) = HighPrecisionMoney.fromPreciseAmount(123456L, 1, Euro, None)
+
+      errors.toList must be (List("fractionDigits must be > 2 (default fraction digits defined by currency EUR)."))
+    }
+
+    it("should validate fractionDigits (max)") {
+      val Invalid(errors) = HighPrecisionMoney.fromPreciseAmount(123456L, 100, Euro, None)
+
+      errors.toList must be (List("fractionDigits must be <= 20."))
+    }
+
+    it("should validate centAmount") {
+      val Invalid(errors) = HighPrecisionMoney.fromPreciseAmount(123456L, 4, Euro, Some(1))
+
+      errors.toList must be (List("centAmount must be correctly rounded preciseAmount (a number between 1234 and 1235)."))
```

Comment:
Can you explain this case? I don't get it.

Reply:
It's related to the issue we discussed previously. `centAmount` is just a rounded of `preciseAmount` for HP Money. Since at the moment it is possible to provide `centAmount` value directly, I wanted to make sure that `ceiling(preciseAmount) >= centAmount >= floor(preciseAmount)`.